### PR TITLE
Bump solc version to 0.5.0

### PR DESCRIPTION
**Pull request by JChanceHud**
This pull request bumps the `solc` version in the `truffle` and `truffle-compile` packages to [`0.5.0`](https://www.npmjs.com/package/solc) which was released a day ago.

I also added a node script `solc-bump` that updates dependent subpackages in `./packages` to `solc@latest`.

To get only the package version updates cherry-pick commit 5b7c6f9f0f0d23eb82814cbe96e4da47ddd8099b.

Thanks!

### CODE REVIEW

**Comment by gnidan** [comment]
!!! Awesome.

Reviewing now.

**Comment by JChanceHud** [comment]
Actually this solidity release seems to have quite a bit of [changes](https://solidity.readthedocs.io/en/latest/050-breaking-changes.html). This may need to be a more extensive refactor than a simple bump :/

File: scripts/solc-bump.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const fs = require('fs');
+const { execSync } = require('child_process');
+const readline = require('readline');
+
+const PACKAGES_DIR = `${__dirname}/../packages`;
```
Comment (gnidan) [inline scripts/solc-bump.js]: Should use `path.resolve()` for this to make it more cross-platform.

File: scripts/solc-bump.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  });
+}
+
+/**
+ * Use child_process to run `npm install` in `packageName` synchronously
+ **/
+function updatePackages(packageNames) {
+  packageNames.forEach(packageName => {
+    console.log(`Updating ${packageName}`);
+    execSync('npm install solc@latest', {
```
Comment (gnidan) [inline scripts/solc-bump.js]: Are requirements for this script to save the value to the `package.json`? It seems that you would need `--save` for that.

File: scripts/solc-bump.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  const packages = fs.readdirSync(PACKAGES_DIR);
+  return packages.filter(packageName => {
+    // parse the package.json file for `packageName`
+    const packagePath = `${PACKAGES_DIR}/${packageName}/package.json`;
+    const packageJson = JSON.parse(fs.readFileSync(packagePath, 'utf8'));
+
+    // operate if the solc dependency exists
+    return packageJson
+      && packageJson.dependencies
+      && !!packageJson.dependencies['solc'];
```
Comment (gnidan) [inline scripts/solc-bump.js]: We might want to include `devDependencies` here; any thoughts?

**Comment by gnidan** [comment]
Related, waiting for Travis: #1434

File: scripts/solc-bump.js (hunk 1 above)
Comment (JChanceHud) [reply scripts/solc-bump.js]: Good call, updated

File: scripts/solc-bump.js (hunk 2 above)
Comment (JChanceHud) [reply scripts/solc-bump.js]: npm install actually saves to dependencies by default, but I added explicity `--save` and `--save-dev` with `devDependencies` support

File: scripts/solc-bump.js (hunk 3 above)
Comment (JChanceHud) [reply scripts/solc-bump.js]: Agreed, updated and refactored a bit

**Pull request merged**